tsl/test/expected/gapfill.out

```diff
@@ -1843,14 +1843,39 @@ FROM metrics_int m1 INNER JOIN metrics_int m2 ON m1.time=m2.time
 WHERE m1.time >=0 AND m1.time < 2
 GROUP BY 1;
 ERROR:  invalid time_bucket_gapfill argument: could not infer start boundary from WHERE clause
+-- test outer join with constraints in join condition
+-- not usable as start/stop
+SELECT
+  time_bucket_gapfill(1,m1.time)
+FROM metrics_int m1 LEFT OUTER JOIN metrics_int m2 ON m1.time=m2.time AND m1.time >=0 AND m1.time < 2
+GROUP BY 1;
+ERROR:  invalid time_bucket_gapfill argument: could not infer start boundary from WHERE clause
```

Comment:
This is an odd error message and it doesn't follow the error style guide. Besides, I cannot figure out what it means. Why is it saying invalid gapfill argument and then complaining about the WHERE clause, which neither is an argument to gapfill, nor exists in the query?

Maybe it should just say: errmsg: `could not infer start boundary for gapfill`. errhint: `A gapfill query requires a start boundary on "time" in the WHERE clause.`

Reply:
A gapfill query requires a start and stop arguments, in the first version you had to supply those as arguments, we later added code to infer start and stop from the WHERE-clause however it can only be infered when additional constraints are satisifed (expression needs to be immutable). So the error means we couldnt infer start from the WHERE clause and it needs to be supplied as argument to the time_bucket_gapfill call (or WHERE clause needs to be rewritten so it can be infered)